app/models/account.rb

```diff
@@ -5,10 +5,28 @@
 # Explicit load of Account model from has_accounts gem.
 require HasAccounts::Railtie.config.root.join('app', 'models', 'account')
 
+class Account::AmbiguousTag < StandardError;
+  def initialize(tag, count)
+    @tag = tag
+    @count = count
+  end
+
+  def to_s
+    "Given tag '#{@tag}' is ambiguous, found #{@count} records"
+  end
+end
+
 # Reopen class do add decorations
 Account.class_eval do
   # Attachments
   # ===========
   has_many :attachments, :as => :reference
   accepts_nested_attributes_for :attachments, :reject_if => proc { |attributes| attributes['file'].blank? }
+
+  def self.find_by_tag(tag)
```

Comment:
I think this would be a valuable addition to the HasAccount gem itself. Will add it there and notify you. (Including the Error).